Formula/openmodelica.rb

```diff
@@ -0,0 +1,94 @@
+class Openmodelica < Formula
+  desc "Open-source modeling and simulation tool"
+  homepage "https://openmodelica.org/"
+  # GitHub's archives lack submodules, must pull:
+  url "https://github.com/OpenModelica/OpenModelica.git",
+    revision: "11fcab4f2d6895f2db073572b2bff1a43177313f", tag: "v1.16.5"
+  license "GPL-3.0-only"
+  head "https://github.com/OpenModelica/OpenModelica.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build # for gfortran
+  depends_on "gnu-sed" => :build
+  depends_on "libtool" => :build
+  depends_on "openjdk" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "boost"
+  depends_on "gettext"
+  depends_on "hdf5"
+  depends_on "hwloc"
+  depends_on "lp_solve"
+  depends_on "omniorb"
+  depends_on "openblas"
+  depends_on "qt"
+  depends_on "readline"
+  depends_on "sundials"
+
+  uses_from_macos "curl"
+  uses_from_macos "expat"
+  uses_from_macos "ncurses"
+
+  # fix omplot, omnotebook, & omoptim until next release (see OpenModelica #7240)
+  patch :DATA
+
+  def install
+    ENV["LDFLAGS"] = "-L#{Formula["openblas"].opt_lib}"
+    ENV["CPPFLAGS"] = "-I#{Formula["openblas"].opt_include}"
+    args = %W[--prefix=#{prefix}
+              --disable-debug
+              --disable-modelica3d
+              --with-cppruntime
+              --with-hwloc
+              --with-lapack=-lopenblas
+              --with-omlibrary=core
+              --with-omniORB]
+
+    system "autoconf"
+    system "./configure", *args
+    # omedit needs omparser, which fails with: unsupported option '-static-libgcc'
+    # omshell needs QTWebKit: #19391 & #19438
+    # omnotebook, omoptim fails with: "No targets.  Stop."
+    # omsens_qt fails with: "OMSens_Qt is not supported on MacOS"
+    system "make", "omc", "omlibrary-core"
+    system "make", "omplot"
+    system "make", "omsimulator"
+    prefix.install Dir["build/*"]
+  end
+
+  test do
+    system "#{bin}/omc", "--version"
+    system "#{bin}/OMPlot --help; test \$? -eq 1" # help is the only nonblocking flag
```

Comment:
Their `--help` flag produces an exit code of `1`? Weird.

Reply:
Yeah it's sad… no args also exits `1`, everything else blocks.

Reply:
How about
```ruby
assert_match output, shell_output("#{bin}/OMPlot --help", 1)
```
where `output` is a string that matches any substring of the output of `OMPlot --help`?

Reply:
Fixed for a later revision.  But I have no idea why the `:exist` test is failing.  It succeeds on my machine:
```
Error: openmodelica: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected #<Pathname:/private/tmp/openmodelica-test-20210303-4951-75b3ta/test_res.mat> to be exist?.
```

Reply:
I'd still like to see a call to `OMPlot` in the test, so please include something like the one I suggested above, rather than just commenting that part of the test out.

Reply:
Oh I disabled the OMPlot build, as it requires patch 7240, so it won't be present until 1.16.6 or a revision.  `omc` is the core component.